platform/gce.go

```diff
@@ -44,10 +46,11 @@ type GCEOptions struct {
 }
 
 type gceCluster struct {
-	api      *compute.Service
-	sshAgent *network.SSHAgent
-	conf     *GCEOptions
-	machines map[string]*gceMachine
+	api        *compute.Service
+	sshAgent   *network.SSHAgent
+	conf       *GCEOptions
+	machines   map[string]*gceMachine
+	sync.Mutex // protects only against NewMachine being called in parallel
```

Comment:
I don't know what's idiomatic for golang, but it's generally accepted to "lock data, not code", so in this case the lock protects the machines map, I'd expect the code to consistently wrap access to the machines map in a lock/unlock sequence.  The comment says the lock protects code, not data, which isn't a good sign.  The comment also doesn't clearly express which members of the structure are protected by the lock.  So at a least, the comment should be updated.

Imagine a future reader investigating a potential race condition/crash, and they suspect there's a data structure not protected with a lock where needed.  Since the code has inconsistently serialized access to this map, the exceptions where there's no locking look like obvious bugs at a glance, until further scrutiny may show that those code paths aren't entered concurrently like the ones with the lock.  What's the harm in holding the lock everywhere this shared data structure is accessed?  It's not performance-sensitive code, so just protect the shared data everywhere it's accessed.